src/Bicep.Core/Registry/Oci/OciManifest.cs

```diff
@@ -8,9 +8,9 @@ namespace Bicep.Core.Registry.Oci
 {
     public class OciManifest
     {
-        // TODO: Add top-level annotations
-        public OciManifest(int schemaVersion, string? artifactType, OciDescriptor config, IEnumerable<OciDescriptor> layers)
+        public OciManifest(int schemaVersion, string? artifactType, OciDescriptor config, IEnumerable<OciDescriptor> layers, OciAnnotations? annotations = null)
```

Comment:
>OciAnnotations [](http://example.com/codeflow?start=125&length=14)

Can you take a look at how I modeled annotations on the OCIDescriptor? We should be consistent with that. (This class is meant to be a generic representation of any OCI manifest.)

Reply:
Ah I thought we modeled OciManifest in this repo close to https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Generated/Models/OciManifest.cs. And I modeled OciAnnotations similar to https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Generated/Models/OciAnnotations.cs.

For consistency, I will go ahead and change this to below:
```
        public ImmutableDictionary<string, string> Annotations { get; }
```

Reply:
Makes sense.

I'm not really sure why the SDK is doing that. They might be customizing serialization somehow, but not sure.

Reply:
Updated here:
https://github.com/Azure/bicep/commit/63d578fb3a0e0a3ec200984176f38dad740d79e5

Thanks!